braindecode/models/sinc_shallow.py

```diff
@@ -0,0 +1,340 @@
+import math
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from einops.layers.torch import Rearrange
+
+from braindecode.models.base import EEGModuleMixin
+
+
+class SincShallowNet(EEGModuleMixin, nn.Module):
+    """Sinc-ShallowNet from Borra, D et al (2020) [borra2020]_.
+
+    .. figure:: https://ars.els-cdn.com/content/image/1-s2.0-S0893608020302021-gr2_lrg.jpg
+        :align: center
+        :alt: SincShallowNet Architecture
+
+    The Sinc-ShallowNet architecture has these fundamental blocks:
+
+    1. **Block 1: Spectral and Spatial Feature Extraction**
+        - *Temporal Sinc-Convolutional Layer*:
+            Uses parametrized sinc functions to learn band-pass filters,
+            significantly reducing the number of trainable parameters by only
+            learning the lower and upper cutoff frequencies for each filter.
+       - *Spatial Depthwise Convolutional Layer*:
+            Applies depthwise convolutions to learn spatial filters for
+            each temporal feature map independently, further reducing
+            parameters and enhancing interpretability.
+       - *Batch Normalization*
+
+    2. **Block 2: Temporal Aggregation**
+        - *Activation Function*: ELU
+        - *Average Pooling Layer*: Aggregation by averaging spatial dim
+        - *Dropout Layer*
+        - *Flatten Layer*
+
+
+    3. **Block 3: Classification**
+        - *Fully Connected Layer*: Maps the feature vector to n_outputs.
+
+    **Implementation Notes:**
+
+    - The sinc-convolutional layer initializes cutoff frequencies uniformly
+        within the desired frequency range and updates them during training while
+        ensuring the lower cutoff is less than the upper cutoff.
+
+    Parameters
+    ----------
+    num_temp_filters : int
```

Comment:
```suggestion
    num_time_filters : int
```
`temp_filters` is misleading, might be understood as temporary. Better use `time` or `temporal`.